src/ripple/app/paths/PathRequest.cpp

```diff
@@ -690,10 +689,12 @@ Json::Value PathRequest::doUpdate(
 
     Json::Value& jvArray = (newStatus[jss::alternatives] = Json::arrayValue);
     if (! findPaths(cache, iLevel, jvArray))
+    {
+        bLastSuccess = false;
         newStatus = rpcError(rpcINTERNAL);
-
-    bLastSuccess = jvArray.size();
-    iLastLevel = iLevel;
+    }
+    else
+        bLastSuccess = jvArray.size() != 0;
 
```

Comment:
Also, I'm not 100% sure if the change of `iLastLevel` to `iLevel` and the removal of the local variable is safe. Don't we call this recursively?